astlib/migrate_501_500.ml

```diff
@@ -262,6 +318,44 @@ and copy_value_binding :
             (List.rev locally_abstract_univars)
         in
         (pvb_pat, pvb_expr)
+    | Some (Pvc_constraint { locally_abstract_univars = []; typ }), _ ->
+        let typ = copy_core_type typ in
+        let pvb_pat =
+          {
+            pvb_pat with
+            ppat_attributes = [];
+            ppat_desc = Ast_500.Parsetree.Ppat_constraint (pvb_pat, typ);
+          }
```

Comment:
This is a ghost node in 5.0.0 with the right end location extended to the right end location of typ.